src/main/java/org/folio/rest/impl/PutOrdersByIdHelper.java

```diff
@@ -0,0 +1,113 @@
+import java.util.concurrent.CompletableFuture;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.log4j.Logger;
+import org.folio.orders.rest.exceptions.HttpException;
+import org.folio.rest.jaxrs.model.CompositePurchaseOrder;
+import org.folio.rest.jaxrs.resource.OrdersResource.PutOrdersByIdResponse;
+import org.folio.rest.tools.client.interfaces.HttpClientInterface;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.json.JsonObject;
+import me.escoffier.vertx.completablefuture.VertxCompletableFuture;
+
+public class PutOrdersByIdHelper {
+  private static final Logger logger = Logger.getLogger(DeleteOrdersByIdHelper.class);
+
+  private final HttpClientInterface httpClient;
+  private final Context ctx;
+  private final Handler<AsyncResult<javax.ws.rs.core.Response>> asyncResultHandler;
+  private final Map<String, String> okapiHeaders;
+
+  public PutOrdersByIdHelper(HttpClientInterface httpClient, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context ctx) {
+    Map<String, String> customHeader = new HashMap<>();
+    customHeader.put(HttpHeaders.ACCEPT.toString(), "application/json, text/plain");
+    httpClient.setDefaultHeaders(customHeader);
+
+    this.httpClient = httpClient;
+    this.okapiHeaders = okapiHeaders;
+    this.ctx = ctx;
+    this.asyncResultHandler = asyncResultHandler;
+  }
+  
+  /*
+   * Handle update order by doing Delete and Post
+   */
+  public CompletableFuture<Void> updateOrder(String id, String lang, CompositePurchaseOrder compPO,
+      Context vertxContext) {
+    CompletableFuture<Void> future = new VertxCompletableFuture<>(ctx);
+
+    PostOrdersHelper postHelper = new PostOrdersHelper(httpClient, okapiHeaders, asyncResultHandler, vertxContext);
+    DeleteOrdersByIdHelper delHelper = new DeleteOrdersByIdHelper(httpClient, okapiHeaders, asyncResultHandler,
+        vertxContext);
+
+    delHelper.deleteOrder(id, lang)
+    .thenRun(() -> {
+        compPO.setId(id);
+        postHelper.createPOandPOLines(compPO)
+        .thenAccept(withCompPO -> {
+          
+          logger.info("Applying Funds...");
+          postHelper.applyFunds(withCompPO)
+          .thenAccept(withFunds -> {
+              
+            logger.info("Updating Inventory...");
+            postHelper.updateInventory(withFunds)
+            .thenAccept(withInventory -> {
+              
+              logger.info("Successfully Placed Order: " + JsonObject.mapFrom(withCompPO).encodePrettily());
+              httpClient.closeClient();
+              javax.ws.rs.core.Response response = PutOrdersByIdResponse.withNoContent();
+              AsyncResult<javax.ws.rs.core.Response> result = Future.succeededFuture(response);
+              asyncResultHandler.handle(result);
+            })
+            .exceptionally(postHelper::handleError);
```

Comment:
you don't want to call the postHelper's handleError method, you want to call this helper's handleError method.